x-pack/metricbeat/module/mssql/performance/performance.go

```diff
@@ -112,13 +117,19 @@ WHERE  counter_name = 'SQL Compilations/sec'
 		row.instanceName = strings.TrimSpace(row.instanceName)
 		row.objectName = strings.TrimSpace(row.objectName)
 
-		if row.counterName == "Buffer cache hit ratio" {
-			mapStr[row.counterName] = fmt.Sprintf("%v", float64(*row.counterValue)/100)
+		if row.counterName == BufferCacheHitRatio {
+			buffer_cache_hit_ratio = *row.counterValue
+		} else if row.counterName == BufferCacheHitRatioBase {
+			buffer_cache_hit_ratio_base = *row.counterValue
 		} else {
 			mapStr[row.counterName] = fmt.Sprintf("%v", *row.counterValue)
 		}
 	}
 
+	if buffer_cache_hit_ratio_base != 0 {
+		mapStr[BufferCacheHitRatio] = fmt.Sprintf("%v", float64(buffer_cache_hit_ratio/buffer_cache_hit_ratio_base))
```

Comment:
After some debugging I think I found the issue:

We need to convert both values to `float64` before performing the division to preserve the fractional part.

```suggestion
		mapStr[BufferCacheHitRatio] = fmt.Sprintf("%v", float64(buffer_cache_hit_ratio)/float64(buffer_cache_hit_ratio_base))
```